File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.py
from collections import defaultdict

def solution(strArr):
    answer = defaultdict(int)
    
#     for i in range(31):
#         answer[i] = 0

#     for str in strArr:
#         key = len(str)
#         if(key in answer):
#             answer[key] += 1
#         else:
#             answer[key] = 1
            
    for str in strArr:
        key = len(str)
        answer[key] += 1

    return max(answer.values())

from collections import defaultdict

File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.java
import java.util.*;

class Solution {
    public int solution(String[] strArr) {
        Map<Integer, Integer> answer = new HashMap<>();
        
        for(String str: strArr){
            int key = str.length();
            
            if(answer.containsKey(key)){
                answer.put(key, answer.get(key) + 1);
            }else{
                answer.put(key, 1);
            }
        }
        
        int max = 0;
        for(int count : answer.values()){
            max = Math.max(max,count);
        }
        
        return max;
    }
}